go-controller/pkg/cluster/gateway_localnet.go

```diff
@@ -294,17 +294,24 @@ func localnetNodePortWatcher(ipt util.IPTablesHelper, wf *factory.WatchFactory)
 	return err
 }
 
-func cleanupLocalnetGateway() error {
+func cleanupLocalnetGateway(physnet string) error {
 	// get bridgeName from ovn-bridge-mappings.
 	stdout, stderr, err := util.RunOVSVsctl("--if-exists", "get", "Open_vSwitch", ".",
 		"external_ids:ovn-bridge-mappings")
 	if err != nil {
 		return fmt.Errorf("Failed to get ovn-bridge-mappings stderr:%s (%v)", stderr, err)
 	}
-	bridgeName := strings.Split(stdout, ":")[1]
-	_, stderr, err = util.RunOVSVsctl("--", "--if-exists", "del-br", bridgeName)
-	if err != nil {
-		return fmt.Errorf("Failed to ovs-vsctl del-br %s stderr:%s (%v)", bridgeName, stderr, err)
+	bridgeMappings := strings.Split(stdout, ",")
+	for _, bridgeMapping := range bridgeMappings {
+		m := strings.Split(bridgeMapping, ":")
+		if physnet == m[0] {
+			bridgeName := m[1]
+			_, stderr, err = util.RunOVSVsctl("--", "--if-exists", "del-br", bridgeName)
+			if err != nil {
+				return fmt.Errorf("Failed to ovs-vsctl del-br %s stderr:%s (%v)", bridgeName, stderr, err)
+			}
+		}
```

Comment:
You can break after L312, since there will no more ovn_bridge_mappings for that physicalNetworkName, right?